pkg/cloud/azureprovider.go

```diff
@@ -847,127 +858,378 @@ func (az *Azure) DownloadPricingData() error {
 	allPrices := make(map[string]*AzurePricing)
 
 	for _, v := range *result.Meters {
-		meterName := *v.MeterName
-		meterRegion := *v.MeterRegion
-		meterCategory := *v.MeterCategory
-		meterSubCategory := *v.MeterSubCategory
-
-		region, err := toRegionID(meterRegion, regions)
+		pricings, err := convertMeterToPricings(v, regions, baseCPUPrice)
 		if err != nil {
+			log.Warnf("converting meter to pricings: %s", err.Error())
 			continue
```

Comment:
do you think an error in this method is serious enough to return it immediately?

Reply:
I was keeping this behaviour the same as it is for the ratecard handling code for consistency, but I'm alright making this an error that stops the pricesheet processing if you'd prefer.

Reply:
your call, sounds like you considered the tradeoffs already. Just wanted to have a discussion!